virttest/test_setup.py

```diff
@@ -2079,3 +2081,46 @@ def disable_smt(params=None):
                 logging.debug("SMT turned off successfully")
             except process.CmdError, info:
                 raise exceptions.TestSetupFail("VM can not be started :%s" % info)
+
+
+class LibvirtdDebugLog(object):
+    """
+    Enable libvirtd log for testcase incase
+    with the use of param "enable_libvirtd_debug_log",
+    with additional params log level("libvirtd_debug_level")
+    and log file path("libvirtd_debug_file") can be controlled.
+    """
+    def __init__(self, env, test, log_level=1, log_file=""):
+        """
+        initialize variables
+
+        :param env: ENV object
+        :param test: Test object
+        :param level: debug level for libvirtd log
+        :param debug_file: debug file path
+        """
+        self.env = env
+        self.log_level = log_level
+        self.log_file = log_file
+        self.test = test
+        self.libvirtd = utils_libvirtd.Libvirtd()
+        self.libvirtd_conf = utils_config.LibvirtdConfig()
+        self.backupfile = "%s.bkup.debuglog" % self.libvirtd_conf.conf_path
+
+    def enable(self):
+        """ Enable libvirtd debug log """
+        if not self.log_file or not os.path.isfile(self.log_file):
+            self.log_file = utils_misc.get_path(self.test.debugdir,
+                                                "libvirtd.log")
+        logging.debug("libvirtd debug log stored in: %s", self.log_file)
+        self.libvirtd_conf["log_level"] = self.log_level
+        self.libvirtd_conf["log_outputs"] = '"1:file:%s"' % self.log_file
+        self.libvirtd.restart()
+        fd = open(self.backupfile, "w")
+        fd.write(self.libvirtd_conf.backup_content)
```

Comment:
it is better to backup first and then to change the `log_level` and `log_outputs`